File: src/main/java/net/internetisalie/lunar/lang/psi/LuaElementType.java
package net.internetisalie.lunar.lang.psi;

import com.intellij.psi.tree.IElementType;
import net.internetisalie.lunar.LuaLanguage;

public class LuaElementType extends IElementType {
    private final String name;

    public LuaElementType(String name) {
        super(name, LuaLanguage.INSTANCE);
        this.name = name;
    }

    public String toString() {
        return name;
    }
}

File: src/main/kotlin/net/internetisalie/lunar/lang/psi/LuaElementType.kt
package net.internetisalie.lunar.lang.psi

import com.intellij.psi.tree.IElementType
import net.internetisalie.lunar.lang.LuaLanguage

class LuaElementType(private val name: String) : IElementType(name, LuaLanguage) {
    override fun toString(): String {
        return name
    }
}
